src/webview/leetCodePreviewProvider.ts

```diff
@@ -1,90 +1,44 @@
 // Copyright (c) jdneo. All rights reserved.
 // Licensed under the MIT license.
 
-import { commands, Disposable, ExtensionContext, ViewColumn, WebviewPanel, window } from "vscode";
+import { commands, ViewColumn } from "vscode";
 import { leetCodeExecutor } from "../leetCodeExecutor";
 import { IProblem } from "../shared";
+import { ILeetCodeWebviewOption, LeetCodeWebview } from "./LeetCodeWebview";
 import { markdownEngine } from "./markdownEngine";
 
-class LeetCodePreviewProvider implements Disposable {
+class LeetCodePreviewProvider extends LeetCodeWebview {
 
-    private context: ExtensionContext;
     private node: IProblem;
-    private panel: WebviewPanel | undefined;
-
-    public initialize(context: ExtensionContext): void {
-        this.context = context;
-    }
+    private description: IDescription;
 
     public async show(node: IProblem): Promise<void> {
-        // Fetch problem first before creating webview panel
         const descString: string = await leetCodeExecutor.getDescription(node);
-
         this.node = node;
-        if (!this.panel) {
-            this.panel = window.createWebviewPanel("leetcode.preview", "Preview Problem", ViewColumn.One, {
-                enableScripts: true,
-                enableCommandUris: true,
-                enableFindWidget: true,
-                retainContextWhenHidden: true,
-                localResourceRoots: markdownEngine.localResourceRoots,
-            });
+        this.description = this.parseDescription(descString, node);
```

Comment:
We can put this line into `getWebviewContent()` to avoid undefined error in the future.

Reply:
The question is, `getWebviewContent()` accepts no arguments (because it will be called in the base class, and there's no way we can know the exact arguments there), so how can we get `descString` and `solutionString`?

I think the proper logic chain is:

---
`getWebviewContent()` binds a html string to `this.panel`
↓
so `getWebviewContent()` will must be called when `this.panel` exists
↓
so `this.panel` should be initialized after `this.description`'s existence is made sure
↓
`this.panel`'s only way to be initialized is by calling `showWebviewInternal()`
↓
so when `this.description` is set before `showWebviewInternal()`, `getWebviewContent()` will never results in undefined error.

---

This is exactly what following code does:
```ts
this.description = this.parseDescription(await leetCodeExecutor.getDescription(node), node);
this.node = node;
if (this.showWebviewInternal()) {
    this.panel.title = `${node.name}: Preview`;
    this.panel.reveal(ViewColumn.One);
}
```

So, the only way to cause such problem is when a new way to initialize `this.panel` is added. Then, it is the adder to ensure `this.panel` is initialized when all it needs is set.

Reply:
I see. Let me think more about this. Just leave the current implementation without change.